### PR TITLE
GEO_SHAPE typed columns must compute generated expressions

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**

### CODE REVIEW

File: server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java [hunk 1]
```diff
@@ -978,4 +978,38 @@ public void test_geo_shape_can_be_not_null() {
...
+        );
+
+        execute("insert into tbl (id) values (1)");
+        execute("refresh table tbl");
+        execute("select g from tbl where id = 1");
+
+        assertThat(
+            TestingHelpers.printedTable(response.rows()),
+            is("{coordinates=[[[5.0, 5.0], [5.0, 30.0], [30.0, 30.0], [30.0, 5.0], [5.0, 5.0]]], type=Polygon}\n")
+        );
```
Comment (BaurzhanSakhariev) [inline server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java:998]: Not sure whether I need this assertion - probably `SHOW CREATE TABLE` is enough?

`SHOW CREATE TABLE` covers not only gen expression but also geo configs - I anyway need such test as I initially noticed this while working on `ADD COLUMN` (see https://github.com/crate/crate/pull/13238/commits/408ccd6524eb2a7ea0f666b76124d0c800bcda82) and this test covers that all configs are handled in planner and properly merged in a new transport action.

**Comment by BaurzhanSakhariev** [comment]
Hi @seut, can you please take a look? This is pre-requisite for https://github.com/crate/crate/pull/13238/ (test case in this PR was originally added there and not passing currently)

File: server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java (hunk 1 above)
Comment (seut) [reply server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java:998]: Afaik the expression evaluation here does not differ from any other generated expression and such is already tested elsewhere. I think a show table test is enough for this case, but no strong opinion.

File: docs/appendices/release-notes/unreleased.rst [hunk 2]
```diff
@@ -73,4 +73,5 @@ Fixes
 .. stable branch. You can add a version label (`v/X.Y`) to the pull request for
 .. an automated mergify backport.
 
-None
+- Fixed an issue that caused ignoring generated expressions on ``GEO_SHAPE``
+  columns.
```
Comment (seut) [inline docs/appendices/release-notes/unreleased.rst:77]:
```suggestion
- Fixed an issue that caused the generated expressions on columns of type
  ``GEO_SHAPE`` not being evaluated on writes and such being ignored.
```

File: server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java [hunk 3]
```diff
@@ -978,4 +978,38 @@ public void test_geo_shape_can_be_not_null() {
...
 
     }
+
+    @Test
+    public void test_add_geo_shape_column_keeps_geo_settings_and_adds_generated_expression() throws Exception {
+        execute("CREATE TABLE tbl (id INTEGER)");
+
+        execute("ALTER TABLE tbl ADD COLUMN g geo_shape " +
+            "GENERATED ALWAYS AS 'POLYGON (( 5 5, 30 5, 30 30, 5 30, 5 5 ))' " +
+            "INDEX using QUADTREE with (precision='123m', distance_error_pct='0.123')"
```
Comment (seut) [inline server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java:988]: Looks like this is not related to `ADD COLUMN` but also happen if the column was defined on table creation. Thus I'd suggest to use a test which defines it on table creation (and rename the test according to that) to avoid any confusion. Otherwise one could think that this is dedicated to the alter statement.

File: server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java (hunk 1 above)
Comment (BaurzhanSakhariev) [reply server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java]: I agree that we can simplify, removed it

File: server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java (hunk 3 above)
Comment (BaurzhanSakhariev) [reply server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java:988]: Yes, it happens on creation as well. I covered it using `ADD COLUMN` since it was a bug very specific to the new [PR](https://github.com/crate/crate/pull/13238/files) - so even if I fix it here, CREATE would pass but ADD COLUMN would still fail on the ADD COLUMN implementation and idea was to have only 1 test for this issue.

Would it be fine if I use `CREATE` in this PR but add similar test for `ADD COLUMN in another PR?

Comment (seut) [reply server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java]: Ah ok, thanks for this clarification. What about adding a `DocIndexMetadataTest` and `MetadataToASTNodeResolverTest` test case to validate this fix? Would this also validate the `ADD COLUM` scenario?

Comment (BaurzhanSakhariev) [reply server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java]: Updated tests in https://github.com/crate/crate/pull/13254/commits/85c9b8a09a3e3e46e087c43bbd83a9fc19a7cdf0

`ADD COLUMN` would need separate test case as it's basically same bug (forgetitng to extract actual Reference out of GeneratedReference) but in the new ADD COLUMN transport action. But as it's a different scenario I think it would be fine to cover it with another test.

**Comment by BaurzhanSakhariev** [comment]
unrelated: Could not GET 'https://jcenter.bintray.com/com/github/jk1/gradle-license-report/2.1/gradle-license-report-2.1.module'. Received status code 403 from server: Forbidden

Retest this please

File: server/src/test/java/io/crate/analyze/MetadataToASTNodeResolverTest.java [hunk 4]
```diff
@@ -514,4 +514,14 @@ public void test_bit_string_length_is_shown_in_show_create_table_output() throws
...
     }
+
+    @Test
+    public void test_generated_expression_on_geo_shape_in_show_create_table_output() throws Exception {
+        SQLExecutor e = SQLExecutor.builder(clusterService)
+            .addTable("create table t (g geo_shape generated always as 'POLYGON (( 5 5, 30 5, 30 30, 5 30, 5 5 ))')")
+            .build();
+        DocTableInfo table = e.resolveTableInfo("t");
+        CreateTable<?> node = MetadataToASTNodeResolver.resolveCreateTable(table);
+        assertThat(SqlFormatter.formatSql(node), Matchers.containsString("\"g\" GEO_SHAPE GENERATED ALWAYS AS"));
```
Comment (seut) [inline server/src/test/java/io/crate/analyze/MetadataToASTNodeResolverTest.java:525]: Maybe worth to assert on the generated expression as well?

File: server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java [hunk 5]
```diff
@@ -976,6 +976,5 @@ public void test_geo_shape_can_be_not_null() {
                 )
                 """.stripIndent()
         ));
-
```
Comment (seut) [inline server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java:979]: minor: Remove unrelated change

**Review by seut — APPROVED** [review]
Added more minor comments, looks good otherwise.
